tests/functions/module_lookup.test.ts

```diff
@@ -1418,3 +1418,136 @@ describe("INDEX formula", () => {
     expect(evaluateCell("A3", { A3: "=INDEX(A1:B2, 2, 2)", ...grid })).toBe("#BAD_EXPR");
   });
 });
+
+describe("INDIRECT formula", () => {
+  test("Check argument validity", () => {
+    expect(evaluateCell("A1", { A1: "=INDIRECT()" })).toBe("#BAD_EXPR");
+    expect(evaluateCell("A1", { A1: '=INDIRECT("B1", "string")' })).toBe("#ERROR");
+    expect(evaluateCell("A1", { A1: '=INDIRECT("B1", false)' })).toBe("#REF");
+    expect(evaluateCell("A1", { A1: '=INDIRECT("R1C1", true)' })).toBe("#REF");
+    expect(evaluateCell("A1", { A1: '=INDIRECT("wrong_reference")' })).toBe("#REF");
+    expect(evaluateCell("A1", { A1: "=INDIRECT(,false)" })).toBe("#REF");
+  });
+
+  test("INDIRECT detect circular error", () => {
+    const grid = evaluateGrid({
+      A1: "A2",
+      A2: "=INDIRECT(A1)",
+      A3: '=INDIRECT("A3")',
+    });
+    expect(grid.A2).toBe("#CYCLE");
+    expect(grid.A3).toBe("#CYCLE");
+  });
+
+  test("functional test without grid context", () => {
+    const model = new Model();
+    const sheetId = model.getters.getActiveSheetId();
+    setCellContent(model, "A1", "kikoulol");
+    expect(model.getters.evaluateFormula(sheetId, "=INDIRECT()")).toBe("#BAD_EXPR"); // @compatibility: on google sheets, return #N/A
+    expect(model.getters.evaluateFormula(sheetId, '=INDIRECT("A1")')).toBe("kikoulol");
+  });
+
+  test("Using address string as reference (A1 notation)", () => {
+    const grid = evaluateGrid({
+      A1: "1",
+      A2: "A2",
+      A4: '=INDIRECT("A1")',
+      A5: '=INDIRECT("A2")',
+    });
+    expect(grid.A4).toBe(1);
+    expect(grid.A5).toBe("A2");
+  });
+
+  test("Using cell content as reference (A1 notation)", () => {
+    const grid = evaluateGrid({
+      A1: "1",
+      A2: "A1",
+      A3: "A2",
+      A4: "=INDIRECT(A2)",
+      A5: "=INDIRECT(A3)",
+    });
+    expect(grid.A4).toBe(1);
+    expect(grid.A5).toBe("A1");
+  });
+
+  test("Using computation result as reference (A1 notation)", () => {
+    const grid = evaluateGrid({
+      A1: "1",
+      A2: "A2",
+      A4: '=INDIRECT("A"&ROW(B1))',
+      A5: '=INDIRECT(CONCAT("A","2"))',
+    });
+    expect(grid.A4).toBe(1);
+    expect(grid.A5).toBe("A2");
+  });
+
+  test("Reference to a cell as a function argument (A1 notation)", () => {
+    const grid = evaluateGrid({
+      A1: "1",
+      A2: "2",
+      A3: "3",
+      A4: '=MAX(INDIRECT("A1"),0)',
+      A5: '=INDIRECT("A2")*INDIRECT("A3")',
+    });
+    expect(grid.A4).toBe(1);
+    expect(grid.A5).toBe(6);
+  });
+
+  test("Reference to a range as a function argument (A1 notation)", () => {
+    const grid = evaluateGrid({
+      A1: "1",
+      A2: "2",
+      A3: "3",
+      A4: '=SUM(INDIRECT("A1:A3"))',
```

Comment:
what happens with `=INDIRECT("A1:A3")` ? You can add a test, I don't think there is one.

Reply:
also test unbounded ranges